src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c

```diff
@@ -28,45 +37,543 @@ const ActorInit Demo_Syoten_InitVars = {
     (ActorFunc)DemoSyoten_Draw,
 };
 
-#endif
+u8 D_80C177D0[] = {
+    0, 1, 1, 0, 2, 2, 2, 2, 0, 0, 1, 0, 2, 2, 0, 1, 0, 2, 2, 0, 1, 0, 2, 2, 0, 1, 0,
+    2, 2, 0, 1, 0, 1, 2, 2, 1, 2, 2, 0, 0, 0, 1, 1, 0, 2, 2, 2, 2, 0, 0, 1, 0, 2, 2,
+    0, 1, 0, 2, 2, 0, 1, 0, 2, 2, 2, 0, 0, 2, 0, 1, 1, 0, 2, 2, 2, 0, 1, 0, 2, 2, 0,
+};
+
+Color_RGBA8 D_80C17824[] = {
+    { 160, 100, 130, 255 },
+    { 150, 120, 160, 255 },
+    { 130, 90, 150, 255 },
+    { 90, 70, 120, 255 },
+};
+
+Color_RGBA8 D_80C17834[] = {
+    { 110, 50, 80, 255 },
+    { 120, 100, 130, 255 },
+    { 90, 40, 100, 255 },
+    { 60, 50, 100, 255 },
+};
+
+void DemoSyoten_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    DemoSyoten* this = THIS;
+
+    Actor_SetScale(&this->actor, 0.3f);
+
+    this->actionFunc = func_80C16A64;
+    this->unk_3E4 = 0;
+    this->unk_3E6 = 0;
+    this->unk_3DC = NULL;
+    this->unk_3E0 = NULL;
+    this->unk_3F2 = 0;
+    this->unk_3D8 = 1.0f;
+
+    switch (DEMOSYOTEN_GET_F(&this->actor)) {
+        case DEMOSYOTEN_F_0:
+            func_80183430((void*)this->unk_144, (void*)&object_syoten_Blob_001328, (void*)&object_syoten_Blob_00023C,
```

Comment:
Are these void casts needed? It seems more likely that `unk_144` is a struct referenced by a pointer anyway.

Reply:
To suppress warnings they are yeah. UNK_TYPE is s32, this is a char[]. I know it's a struct but I have no idea what kind of struct and it seemed pointless to just make an empty/padded struct when I don't even know how big it is.

Reply:
Use `UNK_PTR` then.